addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+        } else if (thingTypeUID.equals(THING_TYPE_PRODUCTION_INVERTER_THREE)) {
+            this.queryString = SOLARVIEW_REQUEST_PRODUCTION_INVERTER_THREE;
+        } else if (thingTypeUID.equals(THING_TYPE_ENERGY_INJECTION)) {
+            this.queryString = SOLARVIEW_REQUEST_ENERGY_INJECTION;
+        } else if (thingTypeUID.equals(THING_TYPE_ENERGY_IMPORT)) {
+            this.queryString = SOLARVIEW_REQUEST_ENERGY_IMPORT;
+        }
+        logger.trace("initialize(): Solarview Query String set to {}.", queryString);
+
+        Configuration config = getThing().getConfiguration();
+        //
+        // hostname parameter
+        //
+        try {
+            hostname = (String) config.get(PROPERTY_SOLARVIEW_SERVER);
+        } catch (Exception e) {
+            logger.warn("Cannot set hostname parameter.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access Solarview as hostname is invalid.");
+        }
+
+        if (hostname == null) {
+            // let's go for the default
+            hostname = new String(DEFAULTS_SERVERNAME);
+        }
+        //
+        // refreshInterval parameter
+        //
+        try {
+            refreshInterval = (BigDecimal) config.get(PROPERTY_SOLARVIEW_REFRESH_INTERVAL);
+        } catch (Exception e) {
+            logger.warn("Cannot set refresh parameter.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access Solarview as refresh is invalid.");
+        }
+
+        if (refreshInterval == null) {
+            // let's go for the default
+            refreshInterval = new BigDecimal(DEFAULTS_REFRESH_SECS);
+        }
+        logger.trace("Initialized Solarview handler '{}' with configuration: host '{}', refresh interval {}.",
+                getThing().getUID().toString(), hostname, refreshInterval);
+        boolean success = updateEnergyDataFromServer();
+        if (!success) {
+            logger.trace("initialize(): finished with status CONFIGURATION_ERROR.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    ("Can not access information provider (host '" + hostname + "', port " + SERVER_CONNECTION_DSTPORT
+                            + ")."));
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+            logger.trace("initialize(): finished with status ONLINE.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        logger.trace("dispose() called.");
+        refreshJob.cancel(true);
+        super.dispose();
+    }
+
+    /**
+     * Start the periodic information retrieval from the Solarview server.
+     */
+    private void startAutomaticRefresh() {
+        logger.trace("startAutomaticRefresh() called.");
+        refreshJob = scheduler.scheduleAtFixedRate(() -> {
```

Comment:
Nope the code in the codebase does not always follow the most recent guidelines. Most of the guidelines come from progressive insight, so not always its that easy to change all codebase to follow them. But to guard them for new contributions is from reviewing perspective easier. Although I do understand that you might have look how other bindings did things and this becomes a little bit confusing.